crates/ironrdp-svc/src/lib.rs

```diff
@@ -314,67 +369,46 @@ impl StaticChannelSet {
     /// Inserts a [`StaticVirtualChannel`] into this [`StaticChannelSet`].
     ///
     /// If a static virtual channel of this type already exists, it is returned.
-    pub fn insert<T: StaticVirtualChannel + 'static>(
-        &mut self,
-        val: T,
-    ) -> Option<(Box<dyn StaticVirtualChannel>, ChunkProcessor)> {
-        self.channels
-            .insert(TypeId::of::<T>(), (Box::new(val), ChunkProcessor::new()))
-    }
-
-    /// Gets a reference to a [`StaticVirtualChannel`] trait object by looking up its [`TypeId`].
-    pub fn get_by_type_id(&self, type_id: TypeId) -> Option<(&dyn StaticVirtualChannel, &ChunkProcessor)> {
-        self.channels.get(&type_id).map(|(boxed, cp)| (boxed.as_ref(), cp))
-    }
-
-    /// Gets a mutable reference to a [`StaticVirtualChannel`] trait object by looking up its [`TypeId`].
-    pub fn get_by_type_id_mut(
-        &mut self,
-        type_id: TypeId,
-    ) -> Option<(&mut dyn StaticVirtualChannel, &mut ChunkProcessor)> {
-        if let Some((boxed, cp)) = self.channels.get_mut(&type_id) {
-            Some((boxed.as_mut(), cp))
-        } else {
-            None
-        }
+    pub fn insert<T: StaticVirtualChannelProcessor + 'static>(&mut self, val: T) -> Option<StaticVirtualChannel> {
+        self.channels.insert(TypeId::of::<T>(), StaticVirtualChannel::new(val))
+    }
+
+    /// Gets a reference to a [`StaticVirtualChannel`] by looking up its [`TypeId`].
+    pub fn get_by_type_id(&self, type_id: TypeId) -> Option<&StaticVirtualChannel> {
+        self.channels.get(&type_id)
+    }
+
+    /// Gets a mutable reference to a [`StaticVirtualChannel`] by looking up its [`TypeId`].
+    pub fn get_by_type_id_mut(&mut self, type_id: TypeId) -> Option<&mut StaticVirtualChannel> {
+        self.channels.get_mut(&type_id)
     }
 
-    /// Gets a reference to a [`StaticVirtualChannel`] trait object by looking up its [`TypeId`].
-    pub fn get_by_type<T: StaticVirtualChannel + 'static>(
-        &self,
-    ) -> Option<(&dyn StaticVirtualChannel, &ChunkProcessor)> {
+    /// Gets a reference to a [`StaticVirtualChannel`] by looking up its [`TypeId`].
```

Comment:
Technically it’s not the `TypeId` of the `StaticVirtualChannel`, but the one for the `StaticVirtualChannelProcessor`